From fcef534b728b9d2d8c8e97391569013a93bbb7f8 Mon Sep 17 00:00:00 2001
From: ruijie-opensource <opensource@ruijie.com.cn>
Date: Wed, 26 Mar 2025 17:22:47 +0800
Subject: [PATCH] Summary: Split the specified order and send it to the database

Change-Id: Id49c25fbb79556652237f0dc26086f9eb2f2e6c1
---

diff --git a/config/config_mgmt.py b/config/config_mgmt.py
index 4e3115b..52513a8 100644
--- a/config/config_mgmt.py
+++ b/config/config_mgmt.py
@@ -461,7 +461,16 @@
             # Verify in Asic DB,
             self._verifyAsicDB(db=dataBase, ports=delPorts, portMap=if_name_map, \
                 timeout=MAX_WAIT)
-            self.writeConfigDB(addConfigtoLoad)
+            # The sequence cannot be guaranteed when the brk port written to the config db together
+            # port out of order will make err in sdk, so write each port individually 
+            ports = list(portJson['PORT'].keys())
+            for port in ports:
+                singleConfigtoLoad = dict()
+                tmp = dict()
+                tmp[port] = addConfigtoLoad['PORT'][port]
+                singleConfigtoLoad['PORT'] = tmp
+                self.writeConfigDB(singleConfigtoLoad)
+            # self.writeConfigDB(addConfigtoLoad)
 
         except Exception as e:
             self.sysLog(doPrint=True, logLevel=syslog.LOG_ERR, msg=str(e))
